src/Control/Distributed/Process/Node.hs

```diff
@@ -319,11 +322,38 @@ startServiceProcesses node = do
 
 -- | Force-close a local node
 --
--- TODO: for now we just close the associated endpoint
+-- This function first try to brutally kill all processes, this
+-- process is not guaranteed to continue. So this is application
+-- responsibility to either close all it's processes before dying
+-- or being able to gracefully shutdown.
 closeLocalNode :: LocalNode -> IO ()
-closeLocalNode node =
-  -- TODO: close all our processes, surely!?
-  NT.closeEndPoint (localEndPoint node)
+closeLocalNode node = join $ terminateProcesses
+  where
+    terminateProcesses :: IO (IO ())
+    terminateProcesses = do
+      st <- modifyMVar (localState node) (\st -> return (st, st))
+      let mxACPid = case localEventBus node of
+                      MxEventBus pid _ _ _ -> pid
+                      MxEventBusInitialising -> error "terminateLocalProcesses: The given node is not initialized."
```

Comment:
Is reaching this branch a bug, i.e. an impossible branch? Or is this a possible race condition?